presto-main/src/main/java/com/facebook/presto/sql/relational/SqlToRowExpressionTranslator.java

```diff
@@ -878,6 +878,16 @@ protected RowExpression visitNullIfExpression(NullIfExpression node, Context con
                 // if (equal(cast(first as <common type>), cast(second as <common type>)), cast(null as firstType), first)
                 return specialForm(IF, returnType, equal, constantNull(originalFirst.getType()), originalFirst);
             }
+            else if (isNative) { // If both the types are same, but still native execution
```

Comment:
For presto native, coordinator does not know about the worker type. Hence this branch won't be executed for Presto and test fails. How does isNative work here and shall we need to make Presto coordinator recognize worker type? CC: @pranjalssh @mbasmanova